src/components/vitals/ChartPanel.vue

```diff
@@ -0,0 +1,277 @@
+    <KSkeleton
+      v-if="!productVersion5xxChartData"
+      class="chart-skeleton"
+      type="table"
+    />
+    <AnalyticsChart
+      v-else
+      :chart-data="productVersion5xxChartData"
+      :chart-options="errorsByProductVersionsOptions"
+      :chart-title="helpText.chartTitle5xxProductVersion"
+      :legend-position="'bottom'"
+      :tooltip-title="helpText.totalRequests"
+      data-testid="chart-productversion-5xx"
+      synthetics-data-key="chart-productversion-5xx"
+    />
+
+    <!-- 4xx and 5xx by Status Code -->
+    <KSkeleton
+      v-if="!statusCode4xxChartData"
+      class="chart-skeleton"
+      type="table"
+    />
+    <AnalyticsChart
+      v-else
+      :chart-data="statusCode4xxChartData"
+      :chart-options="errors4xxStatusCodeOptions as AnalyticsChartOptions"
+      :chart-title="helpText.chartTitle4xxStatusCode"
+      :legend-position="'bottom'"
+      :tooltip-title="helpText.totalRequests"
+      data-testid="chart-statuscode-4xx"
+      synthetics-data-key="chart-statuscode-4xx"
+    />
+    <KSkeleton
+      v-if="!statusCode5xxChartData"
+      class="chart-skeleton"
+      type="table"
+    />
+    <AnalyticsChart
+      v-else
+      :chart-data="statusCode5xxChartData"
+      :chart-options="errors5xxStatusCodeOptions as AnalyticsChartOptions"
+      :chart-title="helpText.chartTitle5xxStatusCode"
+      :legend-position="'bottom'"
+      :tooltip-title="helpText.totalRequests"
+      data-testid="chart-statuscode-5xx"
+      synthetics-data-key="chart-statuscode-5xx"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, ref, watch } from 'vue'
+import { useI18nStore } from '@/stores'
+import { AnalyticsChart, AnalyticsChartColors, ChartTypes, lookupStatusCodeColor, AnalyticsChartOptions } from '@kong-ui-public/analytics-chart'
+import { TimeseriesQueryTime } from '@kong-ui-public/analytics-utilities'
+import '@kong-ui-public/analytics-chart/dist/style.css'
+import useChartRequest from '@/composables/useChartRequest'
+import useChartQueryBuilder from '@/composables/useChartQueryBuilder'
+
+// Queries
+import {
+  chartQueryTrafficRequests,
+  chartQueryTrafficLatency,
+  chartQueryProductVersions4xx,
+  chartQueryProductVersions5xx,
+  chartQueryStatusCode4xx,
+  chartQueryStatusCode5xx
+} from '@/constants/chartQueries'
+
+const helpText = useI18nStore().state.helpText.analytics
+
+const props = defineProps<{
+  modelValue,
+  appId
+}>()
+
+const selectedTimeframe = computed(() => props.modelValue.timeframe?.value)
+const selectedProductVersions = computed(() => props.modelValue.apiVersions?.value)
+const timeseriesQueryTime = computed(() => new TimeseriesQueryTime(selectedTimeframe.value))
+const productVersionsCacheKey = computed(() => selectedProductVersions.value.map(entry => entry.value).join('-') || '')
+const appCacheKey = computed(() => `${productVersionsCacheKey.value}-${timeseriesQueryTime.value.startMs()}-${timeseriesQueryTime.value.endMs()}`)
```

Comment:
yes, but now we're doing an await of a batch of Promises instead of using swrv.  

If you're asking whether this could cause another set of endless 500s, I can mock that result and test for this case :)

Reply:
Since `swrv` isn't involved, it seems unlikely that would happen... perhaps a better question is, what do you want the cache key to accomplish?  Is it that you want the query to re-fire if the time selection changes?

I added a tech debt ticket for adding a `cacheKey` method to timeframes in order to uniquely identify changes.  In the meantime, I guess this is fine; it'll just issue requests more often than it needs to.  (A correct cache key would need to do different things depending on if the time was relative or absolute.)

Reply:
Correct, the intent to make sure it fires each time the timepicker changes.